vectorstores/qdrant/qdrant_test.go

```diff
@@ -0,0 +1,384 @@
+		qdrant.WithCollectionName(collectionName),
+		qdrant.WithEmbedder(e),
+	)
+	require.NoError(t, err)
+
+	_, err = store.AddDocuments(
+		context.Background(),
+		[]schema.Document{
+			{PageContent: "The color of the house is blue."},
+			{PageContent: "The color of the car is red."},
+			{PageContent: "The color of the desk is orange."},
+			{PageContent: "The color of the lamp beside the desk is black."},
+			{PageContent: "The color of the chair beside the desk is beige."},
+		},
+	)
+	require.NoError(t, err)
+
+	result, err := chains.Run(
+		context.TODO(),
+		chains.NewRetrievalQAFromLLM(
+			llm,
+			vectorstores.ToRetriever(store, 5, vectorstores.WithScoreThreshold(0.8)),
+		),
+		"What colors is each piece of furniture next to the desk?",
+	)
+	require.NoError(t, err)
+
+	require.Contains(t, result, "black", "expected black in result")
+	require.Contains(t, result, "beige", "expected beige in result")
+}
+
+func TestQdrantRetrieverFilter(t *testing.T) {
+	t.Parallel()
+
+	qdrantURL, apiKey, dimension, distance := getValues(t)
+	collectionName := setupCollection(t, qdrantURL, apiKey, dimension, distance)
+
+	llm, err := openai.New()
+	require.NoError(t, err)
+	e, err := embeddings.NewEmbedder(llm)
+	require.NoError(t, err)
+
+	url, err := url.Parse(qdrantURL)
+	require.NoError(t, err)
+	store, err := qdrant.New(
+		qdrant.WithURL(*url),
+		qdrant.WithAPIKey(apiKey),
+		qdrant.WithCollectionName(collectionName),
+		qdrant.WithEmbedder(e),
+	)
+	require.NoError(t, err)
+
+	_, err = store.AddDocuments(
+		context.Background(),
+		[]schema.Document{
+			{PageContent: "The color of the house is blue."},
+			{PageContent: "The color of the car is red."},
+			{PageContent: "The color of the desk is orange."},
+			{PageContent: "The color of the lamp beside the desk is black."},
+			{PageContent: "The color of the chair beside the desk is beige."},
+		},
+	)
+	require.NoError(t, err)
+
+	_, err = store.AddDocuments(
+		context.Background(),
+		[]schema.Document{
+			{
+				PageContent: "The color of the lamp beside the desk is black.",
+				Metadata: map[string]any{
+					"location": "kitchen",
+				},
+			},
+			{
+				PageContent: "The color of the lamp beside the desk is blue.",
+				Metadata: map[string]any{
+					"location": "bedroom",
+				},
+			},
+			{
+				PageContent: "The color of the lamp beside the desk is orange.",
+				Metadata: map[string]any{
+					"location": "office",
+				},
+			},
+			{
+				PageContent: "The color of the lamp beside the desk is purple.",
+				Metadata: map[string]any{
+					"location": "sitting room",
+				},
+			},
+			{
+				PageContent: "The color of the lamp beside the desk is yellow.",
+				Metadata: map[string]any{
+					"location": "patio",
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	filter := map[string]interface{}{
+		"must": []map[string]interface{}{
+			{
+				"key": "location",
+				"match": map[string]interface{}{
+					"value": "patio",
+				},
+			},
+		},
+	}
```

Comment:
With the GRPC client this will slook something like:
```go
&pb.Filter{
	Must: []*pb.Condition{
		{
			ConditionOneOf: &pb.Condition_Field{
				Field: &pb.FieldCondition{
					Key: "location",
					Match: &pb.Match{
						MatchValue: &pb.Match_Keyword{
							Keyword: "patio",
						},
					},
				},
			},
		},
	}
```